src/main.py

```diff
@@ -199,7 +200,37 @@ def commit_changes(changes: t.List[str]) -> CommittedChanges:
     # Moved to detached HEAD
     log.info("Switching to detached HEAD")
     check_call(["git", "-c", "advice.detachedHead=false", "checkout", "HEAD@{0}"])
-    check_call(["git", "commit", "-am", message])
+
+    retry_commit = False
+    try:
+        check_call(["git", "commit", "-am", message])
+    except subprocess.CalledProcessError:
+        retry_commit = True
+        pass
+
+    if retry_commit:
+        log.warning("Committing failed. Falling back to a sanitised config")
```

Comment:
What's the real-world case you're trying to fix here?

Reply:
In those cases I think it is good to commit with a fallback so that uncommitted changes do not get lost